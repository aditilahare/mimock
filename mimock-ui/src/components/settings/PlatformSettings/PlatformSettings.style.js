import tw from 'tailwind-styled-components';

export const SettingsContainer = tw.div`
    flex
    flex-col
    justify-start
    align-middle
    w-full
    mt-8
    mb-4
`;

export const PlatformSettingsHeader = tw.div`
    font-sans
    font-bold
    subpixel-antialiased	
    text-lg
    text-gray-600
    mx-20
    mb-4
`;

export const PlatformSettingsOptions = tw.div`
    block
    w-11/12
    mx-24
`;

export const SwitchComponentWrapper = tw.div`
    flex
    items-center
    align-middle
    my-2
    justify-start
`;

export const SwitchComponentLabel = tw.div`
    font-sans
    subpixel-antialiased	
    text-gray-600
    mx-10
`;

export const SwitchComponentStatusActions = tw.div`
    flex
    items-center
    align-middle
    mx-10
`;

export const StatusLabel = tw.div`
    font-sans
    font-semibold
    text-sm
    mx-4
    ${(props) => (props.$isActive ? 'text-slate-800' : 'text-slate-300')}
`;
