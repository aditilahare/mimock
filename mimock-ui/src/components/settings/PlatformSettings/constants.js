export const constants = {
	testIds: {
		settingsContainer: 'psettings-container',
		settingsHeader: 'psettings-header',
		switchComponentBulk: 'switch-component-bulk',
		switchComponentAutoflush: 'switch-component-autoflush',
	},
	label: {
		settingsHeader: 'Platform Settings',
		switchComponentBulk: 'Bulk Mock Operations',
		switchComponentAutoflush: 'AutoFlush Recycle Bin',
	},
};
