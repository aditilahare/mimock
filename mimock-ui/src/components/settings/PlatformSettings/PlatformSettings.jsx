import React, { useEffect, useState, useRef } from 'react';
import {
	SettingsContainer,
	PlatformSettingsHeader,
	SwitchComponentWrapper,
	SwitchComponentLabel,
	SwitchComponentStatusActions,
	StatusLabel,
	PlatformSettingsOptions,
} from './PlatformSettings.style';
import { updatePlatformSettings } from 'services/mockManagement/updatePlatformSettings.service';
import { Cookies } from 'react-cookie';
import ActivationSwitch from '@material-ui/core/Switch';
import { useIosSwitchStyles } from '@mui-treasury/styles/switch/ios';
import useNotification from 'hooks/useNotification';
import { notificationTypes } from 'constants/notificationConstants';
import { constants } from './constants';
import { globalConstants } from 'constants/globalConstants';

function PlatformSettings() {
	const cookies = new Cookies();
	const iosStyles = useIosSwitchStyles();
	const authCookieRef = useRef('');
	const csrfCookieRef = useRef('');
	const [isExportImportMocksEnabled, setIsExportImportMocksEnabled] =
		useState(false);
	const [isBinCronJobEnabled, setIsBinCronJobEnabled] = useState(false);

	useEffect(() => {
		authCookieRef.current = cookies.get(globalConstants.AUTH_TOKEN_COOKIE_NAME);
		csrfCookieRef.current = cookies.get(globalConstants.XSRF_COOKIE_NAME);
		const platformSettings = cookies.get(
			globalConstants.PLATFORM_SETTINGS_COOKIE_NAME
		);
		if (platformSettings !== undefined) {
			const { isExportImportEnabled, isFlushBinCronEnabled } = platformSettings;
			setIsExportImportMocksEnabled(isExportImportEnabled);
			setIsBinCronJobEnabled(isFlushBinCronEnabled);
		}
	}, []);

	const updateExportImportSettingHandler = (e) => {
		// call update api
		updatePlatformSettings(e.target.checked, isBinCronJobEnabled)
			.then((res) => {
				const { isExportImportEnabled } = res;
				setIsExportImportMocksEnabled(isExportImportEnabled);
				useNotification({
					type: isExportImportEnabled
						? notificationTypes.NOTIFICATION_TYPE_SUCCESS
						: notificationTypes.NOTIFICATION_TYPE_WARNING,
					title: isExportImportEnabled ? 'Enabled' : 'Disabled',
					message: `Bulk operations has been ${
						isExportImportEnabled ? 'enabled' : 'disabled'
					}`,
					animationIn: 'animate__bounceIn',
					animationOut: 'animate__bounceOut',
				});
			})
			.catch((err) => {
				console.log(err);
			});
	};

	const updateFlushBinCronSettingHandler = (e) => {
		// call update api
		updatePlatformSettings(isExportImportMocksEnabled, e.target.checked)
			.then((res) => {
				const { isFlushBinCronEnabled } = res;
				setIsBinCronJobEnabled(isFlushBinCronEnabled);
				useNotification({
					type: isFlushBinCronEnabled
						? notificationTypes.NOTIFICATION_TYPE_SUCCESS
						: notificationTypes.NOTIFICATION_TYPE_WARNING,
					title: isFlushBinCronEnabled ? 'Enabled' : 'Disabled',
					message: `Autoflush recycle bin has been ${
						isFlushBinCronEnabled ? 'enabled' : 'disabled'
					}`,
					animationIn: 'animate__bounceIn',
					animationOut: 'animate__bounceOut',
				});
			})
			.catch((err) => {
				console.log(err);
			});
	};

	const { root, switchBase, checked, thumb, track } = iosStyles;

	const { testIds, label } = constants;

	return (
		<SettingsContainer data-testid={testIds.settingsContainer}>
			<PlatformSettingsHeader data-testid={testIds.settingsHeader}>
				{label.settingsHeader}
			</PlatformSettingsHeader>
			<PlatformSettingsOptions>
				<SwitchComponentWrapper>
					<SwitchComponentLabel>
						{label.switchComponentBulk}
					</SwitchComponentLabel>
					<SwitchComponentStatusActions>
						<ActivationSwitch
							data-testid={testIds.switchComponentBulk}
							classes={{ root, switchBase, checked, thumb, track }}
							checked={isExportImportMocksEnabled}
							value={isExportImportMocksEnabled}
							onChange={(e) => {
								updateExportImportSettingHandler(e);
							}}
						/>
						<StatusLabel $isActive={isExportImportMocksEnabled}>
							{isExportImportMocksEnabled ? 'Active' : 'Inactive'}
						</StatusLabel>
					</SwitchComponentStatusActions>
				</SwitchComponentWrapper>
				<SwitchComponentWrapper>
					<SwitchComponentLabel>
						{label.switchComponentAutoflush}
					</SwitchComponentLabel>
					<SwitchComponentStatusActions>
						<ActivationSwitch
							data-testid={testIds.switchComponentBulk}
							classes={{ root, switchBase, checked, thumb, track }}
							checked={isBinCronJobEnabled}
							value={isBinCronJobEnabled}
							onChange={(e) => {
								updateFlushBinCronSettingHandler(e);
							}}
						/>
						<StatusLabel $isActive={isBinCronJobEnabled}>
							{isBinCronJobEnabled ? 'Active' : 'Inactive'}
						</StatusLabel>
					</SwitchComponentStatusActions>
				</SwitchComponentWrapper>
			</PlatformSettingsOptions>
		</SettingsContainer>
	);
}

export default PlatformSettings;
