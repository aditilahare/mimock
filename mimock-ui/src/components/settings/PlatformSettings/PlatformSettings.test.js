import React from 'react';
import { render, act } from '@testing-library/react';
import PlatformSettings from './PlatformSettings';

jest.mock('react-router-dom');

describe('PlatformSettings', () => {
	it('should render platform settings', async () => {
		let tree;
		await act(async () => {
			tree = await render(<PlatformSettings />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('psettings-container')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
