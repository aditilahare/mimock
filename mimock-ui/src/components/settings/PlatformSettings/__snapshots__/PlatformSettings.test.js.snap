// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformSettings should render platform settings 1`] = `
<div>
  <div
    class="flex flex-col justify-start align-middle w-full mt-8 mb-4"
    data-testid="psettings-container"
  >
    <div
      class="font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20 mb-4"
      data-testid="psettings-header"
    >
      Platform Settings
    </div>
    <div
      class="block w-11/12 mx-24"
    >
      <div
        class="flex items-center align-middle my-2 justify-start"
      >
        <div
          class="font-sans subpixel-antialiased text-gray-600 mx-10"
        >
          Bulk Mock Operations
        </div>
        <div
          class="flex items-center align-middle mx-10"
        >
          <span
            class="MuiSwitch-root IosSwitch-root-1"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiSwitch-switchBase IosSwitch-switchBase-2 MuiSwitch-colorSecondary"
              data-testid="switch-component-bulk"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-10 MuiSwitch-input"
                  type="checkbox"
                  value="false"
                />
                <span
                  class="MuiSwitch-thumb IosSwitch-thumb-3"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track IosSwitch-track-4"
            />
          </span>
          <div
            class="font-sans font-semibold text-sm mx-4 text-slate-300"
          >
            Inactive
          </div>
        </div>
      </div>
      <div
        class="flex items-center align-middle my-2 justify-start"
      >
        <div
          class="font-sans subpixel-antialiased text-gray-600 mx-10"
        >
          AutoFlush Recycle Bin
        </div>
        <div
          class="flex items-center align-middle mx-10"
        >
          <span
            class="MuiSwitch-root IosSwitch-root-1"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiSwitch-switchBase IosSwitch-switchBase-2 MuiSwitch-colorSecondary"
              data-testid="switch-component-bulk"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-10 MuiSwitch-input"
                  type="checkbox"
                  value="false"
                />
                <span
                  class="MuiSwitch-thumb IosSwitch-thumb-3"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track IosSwitch-track-4"
            />
          </span>
          <div
            class="font-sans font-semibold text-sm mx-4 text-slate-300"
          >
            Inactive
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
